modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportGENCC.pm

```diff
@@ -0,0 +1,227 @@
+package Bio::EnsEMBL::Variation::Pipeline::PhenotypeAnnotation::ImportGENCC;
+
+use strict;
+use warnings;
+use base qw/Bio::EnsEMBL::Variation::Pipeline::BaseVariationProcess/;
+
+use Text::CSV_XS;
+use POSIX qw(strftime);
+use File::Path qw(make_path);
+use HTTP::Tiny;
+use Bio::EnsEMBL::Registry;
+
+sub fetch_input {
+  my ($self) = @_;
+
+  my $species = $self->param_required('species');
+  my $pdir    = $self->param_required('pipeline_dir');
+
+  my $default_dir  = $pdir . "/GenCC";
+  my $default_file = $default_dir . "/gencc.csv";
+  my $file         = $self->param('gencc_file') // $default_file;
+
+  # If file is missing/empty, fetch the latest CSV from GenCC
+  unless (-s $file) {
+    make_path($default_dir);
+    my $url  = 'https://search.thegencc.org/download/action/submissions-export-csv';
+    my $http = HTTP::Tiny->new(timeout => 180);
+    my $res  = $http->get($url);
+    $self->throw("GenCC download failed: $res->{status} $res->{reason}") unless $res->{success};
+    open(my $OUT, ">:encoding(UTF-8)", $default_file) or $self->throw("Cannot write $default_file: $!");
+    print $OUT $res->{content};
+    close $OUT;
+    $file = $default_file;
+  }
+
+  -s $file or $self->throw("GenCC input CSV not found or empty after download: $file");
+
+  my $version = $self->param('gencc_version') // strftime("%Y%m%d", localtime);
+
+  $self->param('input_file',    $file);
+  $self->param('gencc_version', $version);
+}
+
+sub run {
+  my ($self) = @_;
+
+  my $species = $self->param_required('species');
+  my $file    = $self->param_required('input_file');
+  my $version = $self->param_required('gencc_version');
+
+  # Default filters to avoid double attribution (override via pipeline param if needed)
+  my %skip = map { $_ => 1 } @{ $self->param('filter_submitters') || [ 'G2P' ] };
+
+  # DB adaptors / handles
+  my $core_dba = $self->get_species_adaptor('core');
+  my $var_dba  = $self->get_species_adaptor('variation');
+
+  my $dbh_core = $core_dba->dbc->db_handle;
+  my $dbh_var  = $var_dba->dbc->db_handle;
+
+  ############################
+  # Ensure source table contains GenCC info
+  ############################
+  my ($source_id) = $dbh_var->selectrow_array("SELECT source_id FROM source WHERE name='GenCC'");
+  if ($source_id) {
+    my $upd = $dbh_var->prepare("UPDATE source SET version=? WHERE source_id=?");
+    $upd->execute($version, $source_id);
+  } else {
+    my $ins = $dbh_var->prepare(q{
+      INSERT INTO source
+        (name, version, description, url, type, somatic_status, data_types)
+      VALUES
+        (?,    ?,       ?,           ?,   NULL, ?,              ?)
+    });
+    $ins->execute(
+      'GenCC',
+      $version,
+      'Gene Curation Coalition (gene-disease validity assertions)',
+      'https://search.thegencc.org/',
+      'germline',
+      'phenotype_feature',
+    );
+    $source_id = $dbh_var->{mysql_insertid};
```

Comment:
There is a method in BasePhenotypeAnnotation.pm to get or add source: [get_or_add_source](https://github.com/Ensembl/ensembl-variation/blob/ac9116d964b33253cb34f727cad8c1c022d28672/modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/BasePhenotypeAnnotation.pm#L446), you should use it here.
Check the G2P example: https://github.com/Ensembl/ensembl-variation/blob/3ca1405d951c131be72e2915f2a2f2d8eca4eaf8/modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportG2P.pm#L149